src/data/vocab_base.h

```diff
@@ -31,6 +39,13 @@ class IVocab {
   virtual Words encode(const std::string& line,
                        bool addEOS = true,
                        bool inference = false) const = 0;
+  
+  virtual Words encodePreservingSource(const string_view &line, 
```

Comment:
Don't put variable names here because clang will complain about unused function arguments.  It should just be `const string_view &, std::vector<string_view> &, ...)

Reply:
Note to self: Body doesn't use variables, which raises warning.